File: core/src/commonMain/kotlin/work/socialhub/kslack/entity/event/HelloEvent.java
package work.socialhub.kslack.entity.event;

/**
 * The hello event is sent when a connection is opened to the message server.
 * This allows a client to confirm the connection has been correctly opened.
 * <p>
 * https://api.slack.com/events/hello
 */
public class HelloEvent implements Event {

    public static final String TYPE_NAME = "hello";

    private final String type = TYPE_NAME;

    public String getType() {
        return this.type;
    }
}

File: core/src/commonMain/kotlin/work/socialhub/kslack/entity/event/HelloEvent.kt
package work.socialhub.kslack.entity.event

/**
 * The hello event is sent when a connection is opened to the message server.
 * This allows a client to confirm the connection has been correctly opened.
 *
 *
 * https://api.slack.com/events/hello
 */
class HelloEvent : Event {
    override val type: String = TYPE_NAME

    companion object {
        const val TYPE_NAME: String = "hello"
    }
}